### PR TITLE
Add video tab

### PR DESCRIPTION
**Pull request by selinanatschke-mms**
# Description

* split up sessions tab, that included session information and video into two separate tabs: browser info and video
* added more information to the browser info tab e.g. session metrics
* added linking between video id and session id if they are available 

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by mreiche — APPROVED** [review]

File: report-ng/app/src/components/method-details/method.ts [hunk 1]
```diff
@@ -154,6 +164,15 @@ export class Method {
                         }
                         break;
                     }
+                    case "video":{
+                        if(methodDetails.sessionContexts.map(context => context.videoId).length > 0){
```
Comment (martingrossmann) [inline report-ng/app/src/components/method-details/method.ts:168]: Please doubecheck this line. The Video tab is shown even no video is available.

**Review by martingrossmann — CHANGES_REQUESTED** [review]

File: report-ng/app/src/components/method-details/method.ts [hunk 2]
```diff
@@ -165,7 +165,8 @@ export class Method {
                         break;
                     }
                     case "video":{
-                        if(methodDetails.sessionContexts.map(context => context.videoId).length > 0){
+                        const contextsWithVideos = methodDetails.sessionContexts.filter(context => context.videoId?.length > 0)
+                        if(contextsWithVideos.length > 0){
                             routeConfig.settings.count = methodDetails.sessionContexts.map(context => context.videoId).length;
```
Comment (mreiche) [inline report-ng/app/src/components/method-details/method.ts:170]: routeConfig.settings.count = contextsWithVideos.length

File: report-ng/app/src/components/method-details/browser-info.html [hunk 3]
```diff
@@ -0,0 +1,82 @@
+<!--
+  ~ Testerra
+  ~
+  ~ (C) 2020, Mike Reiche, T-Systems Multimedia Solutions GmbH, Deutsche Telekom AG
```
Comment (mreiche) [inline report-ng/app/src/components/method-details/browser-info.html:4]: Thanks ;)

File: report-ng/app/src/components/method-details/method.ts [hunk 4]
```diff
@@ -154,6 +164,16 @@ export class Method {
                         }
                         break;
                     }
+                    case "video":{
+                        const contextsWithVideos = methodDetails.sessionContexts.filter(context => context.videoId?.length > 0)
+                        if(contextsWithVideos.length > 0){
+                            routeConfig.settings.count = methodDetails.sessionContexts.map(context => context.videoId).length;
```
Comment (mreiche) [inline report-ng/app/src/components/method-details/method.ts:170]: routeConfig.settings.count = contextsWithVideos.length

**Review by mreiche — CHANGES_REQUESTED** [review]

**Review by mreiche — APPROVED** [review]

**Comment by martingrossmann** [comment]
Looks great for me. But please fix the report test `testT01_checkDisplayedSessionGotContent`. It still tries to find the old Sessions tab.

File: docs/src/images/report-ng-22.png
Comment (martingrossmann) [inline docs/src/images/report-ng-22.png:1]: Why you changed the image of the classes view to method details view? This screenshot is used in report chapter `More on Assertions`

File: docs/src/images/report-ng-collected-assertions-01.png
Comment (martingrossmann) [inline docs/src/images/report-ng-collected-assertions-01.png:1]: This screenshot is used in report chapter `More on Assertions - Collected assertions`. Now we have 2 images of steps view.

**Review by martingrossmann — CHANGES_REQUESTED** [review]

File: report-ng-tests/src/main/java/io/testerra/report/test/pages/report/methodReport/ReportVideoTab.java [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+    public ReportVideoTab(WebDriver driver) {
+        super(driver);
+    }
+
+    public void validateBrowser(String browser) {
+        this.browser.assertThat().text().contains(browser);
+    }
+
+    public void checkSessionId(String id){
+        this.id.assertThat().text().contains(id);
```
Comment (mreiche) [inline report-ng-tests/src/main/java/io/testerra/report/test/pages/report/methodReport/ReportVideoTab.java:52]: Use `isContaining()` or `contains(id).is(true)`

File: report-ng-tests/src/main/java/io/testerra/report/test/pages/report/methodReport/ReportVideoTab.java [hunk 6]
```diff
@@ -0,0 +1,58 @@
...
+    private final UiElement browser = videoCard.find(By.xpath("//li[./span[contains(text(), 'Browser')]]"));
+    @Check
+    private final UiElement video = videoCard.find(By.xpath("//video"));
+
+    public ReportVideoTab(WebDriver driver) {
+        super(driver);
+    }
+
+    public void validateBrowser(String browser) {
+        this.browser.assertThat().text().contains(browser);
```
Comment (mreiche) [inline report-ng-tests/src/main/java/io/testerra/report/test/pages/report/methodReport/ReportVideoTab.java:48]: Use `isContaining()` or `contains(id).is(true)`

**Review by mreiche — DISMISSED** [review]

File: report-ng-tests/src/main/java/io/testerra/report/test/pages/report/methodReport/ReportVideoTab.java [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+    public ReportVideoTab(WebDriver driver) {
+        super(driver);
+    }
+
+    public void validateBrowser(String browser) {
+        this.browser.assertThat().text().isContaining(browser);
+    }
+
+    public void checkSessionId(){
+        this.id.assertThat().displayed();
```
Comment (martingrossmann) [inline report-ng-tests/src/main/java/io/testerra/report/test/pages/report/methodReport/ReportVideoTab.java:52]: Like discussed this check is duplicated. If you want to check the ID, you should take it from Browser info.

**Review by martingrossmann — APPROVED** [review]

**Pull request merged**